core/client/src/main/java/alluxio/client/file/options/CompleteUfsFileOptions.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the “License”). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.client.file.options;
+
+import alluxio.annotation.PublicApi;
+import alluxio.client.ClientContext;
+import alluxio.security.LoginUser;
+import alluxio.thrift.CompleteUfsFileTOptions;
+import alluxio.util.SecurityUtils;
+
+import com.google.common.base.Objects;
+
+import javax.annotation.concurrent.NotThreadSafe;
+
+/**
+ * Options for completing a UFS file. Currently we do not allow users to set arbitrary user and
+ * group options. The user and group will be set to the user login.
+ */
+@PublicApi
+@NotThreadSafe
+public final class CompleteUfsFileOptions {
+  /** The ufs user this file should be owned by. */
+  private final String mUser;
+  /** The ufs group this file should be owned by. */
+  private final String mGroup;
+
+  /**
+   * @return the default {@link CompleteUfsFileOptions}
+   */
+  //TODO(calvin): Move the user group discovery to a utils method
+  public static CompleteUfsFileOptions defaults() {
+    String user = null;
+    if (SecurityUtils.isAuthenticationEnabled(ClientContext.getConf())) {
+      try {
+        user = LoginUser.get(ClientContext.getConf()).getName();
+      } catch (Exception e) {
+        // Fall through to system property approach
+      }
+    }
+    if (user == null) { // Set to system user, or null if there is no system user set
+      String systemUser = System.getProperty("user.name");
+      user = systemUser.isEmpty() ? null : systemUser;
+    }
+    String group = user;
```

Comment:
For now, group is always identical to user.
Can we please add a TODO to utilize user-to-group mapping? I can address it later.